packages/@uppy/utils/src/RateLimitedQueue.ts

```diff
@@ -1,47 +1,72 @@
-function createCancelError (cause) {
+function createCancelError(cause?: string) {
   return new Error('Cancelled', { cause })
 }
 
-function abortOn (signal) {
+function abortOn(
+  this: { abort: (cause: string) => void },
+  signal: AbortSignal,
+) {
   if (signal != null) {
     const abortPromise = () => this.abort(signal.reason)
     signal.addEventListener('abort', abortPromise, { once: true })
-    const removeAbortListener = () => { signal.removeEventListener('abort', abortPromise) }
+    const removeAbortListener = () => {
+      signal.removeEventListener('abort', abortPromise)
+    }
+    // @ts-expect-error don't how how to type this
     this.then?.(removeAbortListener, removeAbortListener)
   }
 
   return this
 }
 
+type Handler = {
+  shouldBeRequeued?: boolean
+  fn: () => (...args: any[]) => Promise<void> | void
+  priority: number
+  abort: (cause?: string) => void
```

Comment:
I think we accept any cause, let's type it as such
```suggestion
  abort: (cause?: unknown) => void
```